ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -110,12 +110,19 @@ public void run(MLAgent mlAgent, Map<String, String> params, ActionListener<Obje
         String memoryId = params.get(MLAgentExecutor.MEMORY_ID);
         String appType = mlAgent.getAppType();
         String title = params.get(MLAgentExecutor.QUESTION);
+        String regenerateInteractionId = params.get(MLAgentExecutor.REGENERATE_INTERACTION_ID);
 
         ConversationIndexMemory.Factory conversationIndexMemoryFactory = (ConversationIndexMemory.Factory) memoryFactoryMap.get(memoryType);
         conversationIndexMemoryFactory.create(title, memoryId, appType, ActionListener.<ConversationIndexMemory>wrap(memory -> {
             memory.getMessages(ActionListener.<List<Interaction>>wrap(r -> {
                 List<Message> messageList = new ArrayList<>();
                 for (Interaction next : r) {
+                    // ignore regenerate interaction question/answer as chat history context to send to LLM
+                    if (next.getId().equals(regenerateInteractionId)) {
+                        // there may have other new interactions happened after this interaction,
```

Comment:
Given the interaction will be deleted before chat agent is executed, is this check required?

Reply:
The deletion happen before response return not start execution.

Reply:
Can executeAgent be called after older interaction is deleted? Ideally chat agent  need not be aware that it is a regenerate request.

Reply:
sounds good, remove the check and delete interaction before actual execution.